### PR TITLE
feat: custom verifier (VF-000)

### PR DESCRIPTION
**Pull request by DecathectZero**
**Fixes or implements VF-000**

### Brief description. What is this change?

removed third party https://www.npmjs.com/package/alexa-verifier-middleware and https://github.com/mreinstein/alexa-verifier
and using the official https://www.npmjs.com/package/ask-sdk-express-adapter instead.

documentation reference: https://developer.amazon.com/en-US/docs/alexa/alexa-skills-kit-sdk-for-nodejs/host-web-service.html#for-web-application-without-express-framework

I opted to use the "For web application without Express framework" documentation because our controller expects to act on the actual request and response instead of getting the whole thing wrapped.

ripped some code patterns from: https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/blob/master/ask-sdk-express-adapter/lib/util/index.ts

### Related PRs

| branch              | PR          |
| ------------------- | ----------- |
| other_pr_production | [link](url) |
| other_pr_master     | [link](url) |

### Checklist

- [ ] title of PR reflects the branch name
- [ ] all commits adhere to conventional commits
- [ ] appropriate tests have been written
- [ ] all the dependencies are upgraded

### CODE REVIEW

File: lib/middlewares/alexa.ts [hunk 1]
```diff
@@ -1,17 +1,59 @@
...
+  req.on('data', (data: string) => {
+    // eslint-disable-next-line no-return-assign
+    return (req.rawBody += data);
+  });
+
+  req.on('end', () => {
+    try {
+      req.body = JSON.parse(req.rawBody);
+    } catch (parseError) {
+      log.error('body parse rejection: %s, %s', parseError, req.rawBody);
```
Comment (jonahsnider) [inline lib/middlewares/alexa.ts:32]: the voiceflow/logger package hasn't been updated here so this still compiles.
the logger accepts a single string argument, ex.
```suggestion
      log.error(`body parse rejection:${parseError}, ${req.rawBody}`);
```

**Review by jonahsnider — APPROVED** [review]

File: lib/middlewares/alexa.ts (hunk 1 above)
Comment (DecathectZero) [reply lib/middlewares/alexa.ts:32]: I see, we should update it everywhere else

**Review by jonahsnider — APPROVED** [review]

**Review by jonahsnider — APPROVED** [review]

File: backend/api/routers/alexa.ts [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
 
-import { BODY_PARSER_SIZE_LIMIT } from '@/backend/constants';
 import { ControllerMap, MiddlewareMap } from '@/lib';
 
 export default (middlewares: MiddlewareMap, controllers: ControllerMap) => {
   const router = express.Router();
 
+  router.use(bodyParser.text({ type: '*/*' }));
   router.use(middlewares.alexa.verifier);
-  router.use(bodyParser.json({ limit: BODY_PARSER_SIZE_LIMIT }));
```
Comment (DecathectZero) [inline backend/api/routers/alexa.ts:11]: the `middlewares.alexa.verifier` does a JSON Parse on the body already, no need for bodyParser

Comment (jonahsnider) [reply backend/api/routers/alexa.ts:11]: might be worth adding a comment to the source

**Review by xavidop — APPROVED** [review]

**Review by jonahsnider — APPROVED** [review]

**Pull request merged**